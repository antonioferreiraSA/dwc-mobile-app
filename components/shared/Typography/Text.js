/* eslint-disable react-native/no-inline-styles */
/**
 * Text component that uses the correct font-family
 * and supports different font weights
 */

import React from 'react';
import { Text as RNText } from 'react-native';
import Colors from '../../../constants/Colors';

function getFontRhythm(props) {
  if (props.XXL) {
    return { fontSize: 30, lineHeight: 40 };
  }
  if (props.XL) {
    return { fontSize: 28, lineHeight: 34 };
  }
  if (props.L) {
    return { fontSize: 22, lineHeight: 28 };
  }
  if (props.M) {
    return { fontSize: 18, lineHeight: 24 };
  }
  if (props.S) {
    return { fontSize: 16, lineHeight: 20 };
  }
  if (props.XS) {
    return { fontSize: 14, lineHeight: 16 };
  }

  // the base font size
  return { fontSize: 18, lineHeight: 24 };
}

export function Text(props) {
  const fontRhythm = getFontRhythm(props);

  let fontWeight = 'Regular';

  if (props.light) {
    fontWeight = 'Light';
  } else if (props.bold) {
    fontWeight = 'Bold';
  }

  return (
    <RNText
      selectable
      {...props}
      style={[
        {
          fontFamily: `NunitoSans-${fontWeight}${props.italic ? 'Italic' : ''}`,
          textAlign: props.center ? 'center' : 'auto',
          color: Colors.lightGray, // ! use theme to determine this
          ...fontRhythm,
        },
        props.style,
      ]}
    >
      {props.children}
    </RNText>
  );
}
