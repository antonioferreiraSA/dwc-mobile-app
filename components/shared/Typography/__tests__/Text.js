import React from 'react';
import { render } from '@testing-library/react-native';
import { Text } from '../Text';

describe('<Text/>', () => {
  test('render text', async () => {
    const { toJSON } = render(<Text>ðŸ”¥</Text>);

    expect(toJSON()).toMatchSnapshot();
  });

  test('render text with a light font', async () => {
    const { toJSON } = render(<Text light>ðŸ”¥</Text>);

    expect(toJSON()).toMatchSnapshot();
  });

  test('render text with a bold font', async () => {
    const { toJSON } = render(<Text bold>ðŸ”¥</Text>);

    expect(toJSON()).toMatchSnapshot();
  });

  test('render text with style', async () => {
    const { toJSON } = render(<Text style={{ fontSize: 30 }}>ðŸ”¥</Text>);

    expect(toJSON()).toMatchSnapshot();
  });

  test('render text with extra props', async () => {
    const { toJSON } = render(
      <Text testID="text-with-some-props" disabled>
        ðŸ”¥
      </Text>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
