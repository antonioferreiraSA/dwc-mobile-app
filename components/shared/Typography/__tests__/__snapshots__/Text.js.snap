// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text/> render text 1`] = `
<Text
  selectable={true}
  style={
    Array [
      Object {
        "color": "#cbd2d6",
        "fontFamily": "NunitoSans-Regular",
        "fontSize": 18,
        "lineHeight": 24,
        "textAlign": "auto",
      },
      undefined,
    ]
  }
>
  ðŸ”¥
</Text>
`;

exports[`<Text/> render text with a bold font 1`] = `
<Text
  bold={true}
  selectable={true}
  style={
    Array [
      Object {
        "color": "#cbd2d6",
        "fontFamily": "NunitoSans-Bold",
        "fontSize": 18,
        "lineHeight": 24,
        "textAlign": "auto",
      },
      undefined,
    ]
  }
>
  ðŸ”¥
</Text>
`;

exports[`<Text/> render text with a light font 1`] = `
<Text
  light={true}
  selectable={true}
  style={
    Array [
      Object {
        "color": "#cbd2d6",
        "fontFamily": "NunitoSans-Light",
        "fontSize": 18,
        "lineHeight": 24,
        "textAlign": "auto",
      },
      undefined,
    ]
  }
>
  ðŸ”¥
</Text>
`;

exports[`<Text/> render text with extra props 1`] = `
<Text
  disabled={true}
  selectable={true}
  style={
    Array [
      Object {
        "color": "#cbd2d6",
        "fontFamily": "NunitoSans-Regular",
        "fontSize": 18,
        "lineHeight": 24,
        "textAlign": "auto",
      },
      undefined,
    ]
  }
  testID="text-with-some-props"
>
  ðŸ”¥
</Text>
`;

exports[`<Text/> render text with style 1`] = `
<Text
  selectable={true}
  style={
    Array [
      Object {
        "color": "#cbd2d6",
        "fontFamily": "NunitoSans-Regular",
        "fontSize": 18,
        "lineHeight": 24,
        "textAlign": "auto",
      },
      Object {
        "fontSize": 30,
      },
    ]
  }
>
  ðŸ”¥
</Text>
`;
